### PR TITLE
Return result directly if available in GeoIPBackend

### PR DESCRIPTION
**Pull request by cmouse**
Do not return CNAME if result can be directly found

### CODE REVIEW

**Comment by cmouse** [comment]
Ready for review

File: modules/geoipbackend/geoipbackend.cc [hunk 1]
```diff
@@ -195,6 +193,20 @@ void GeoIPBackend::lookup(const QType &qtype, const DNSName& qdomain, DNSPacket
   
   format = format2str(format, ip, v6);
 
+  // see if the record can be found
+  if (dom.records.count(format)) { // return static value
+    map<DNSName, vector<DNSResourceRecord> >::iterator i = dom.records.find(format);
+    BOOST_FOREACH(DNSResourceRecord rr, i->second) {
```
Comment (pieterlexis) [inline modules/geoipbackend/geoipbackend.cc]: Could you change this to a C++ 2011 auto for?

**Comment by pieterlexis** [comment]
It looks good, could you add a test to the `regression-tests.nobackend` that queries a direct value and a non-direct value?

**Comment by cmouse** [comment]
GeoIP has it's own test suite in ../modules/geoipbackend/regression-tests/. I can perhaps add it there.

File: modules/geoipbackend/geoipbackend.cc (hunk 1 above)
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: how about I do a code cleanup PR separately? =)

Comment (pieterlexis) [inline modules/geoipbackend/geoipbackend.cc]: :+1:

**Comment by pieterlexis** [comment]
I have a feeling those tests aren't run by travis (the whole geoip-backend isn't at the moment, but these should be tested once we get travis sorted out)?

**Comment by cmouse** [comment]
those tests are ran by travis, but not master.

**Comment by cmouse** [comment]
Added tests and fixed a bug (would be nice to actually run the test suite in travis...)

**Comment by pieterlexis** [comment]
I'm now running some tests in travis to see if we can do just that :)

File: modules/geoipbackend/geoipbackend.cc (hunk 1 above)
Comment (Habbie) [inline modules/geoipbackend/geoipbackend.cc]: Note that that would hurt backporting to 3.4.6 (should we care)

Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: I'll do separate code cleanup, after this.

**Pull request merged**